brave/src/main/java/brave/sampler/Matchers.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2013-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.sampler;
+
+import java.util.Arrays;
+
+/**
+ * Convenience functions to compose matchers for {@link ParameterizedSampler}.
+ *
+ * @see Matcher
+ * @see ParameterizedSampler
+ * @since 5.8
+ */
+public final class Matchers {
+
+  /** @since 5.8 */
+  public static <P> Matcher<P> alwaysMatch() {
+    return (Matcher<P>) Constants.ALWAYS_MATCH;
+  }
+
+  /** @since 5.8 */
+  public static <P> Matcher<P> neverMatch() {
+    return (Matcher<P>) Constants.NEVER_MATCH;
+  }
+
+  enum Constants implements Matcher<Object> {
+    ALWAYS_MATCH {
+      @Override public boolean matches(Object parameters) {
+        return true;
+      }
+
+      @Override public String toString() {
+        return "matchAll()";
+      }
+    },
+    NEVER_MATCH {
+      @Override public boolean matches(Object parameters) {
+        return false;
+      }
+
+      @Override public String toString() {
+        return "neverMatch()";
+      }
+    }
+  }
+
+  /** @since 5.8 */
+  public static <P> Matcher<P> and(Matcher<P>... matchers) {
```

Comment:
I don't know if there is a style in Brave, but if not, would recommend providing a version that takes `Iterable` too, converting an `Iterable` to array in Java is annoying enough that it's nice if libraries do it.

Reply:
no problem.